core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/store/dot-container-list.store.ts

```diff
@@ -0,0 +1,212 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { MenuItem } from 'primeng/api';
+import { DotContainer } from '@models/container/dot-container.model';
+import { ActionHeaderOptions } from '@models/action-header';
+import { DataTableColumn } from '@models/data-table';
+import { ActivatedRoute } from '@angular/router';
+import { pluck, take } from 'rxjs/operators';
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import { DotRouterService } from '@services/dot-router/dot-router.service';
+import { DotPushPublishDialogService } from '@dotcms/dotcms-js';
+
+export interface DotContainerListState {
+    containerBulkActions: MenuItem[];
+    selectedContainers: DotContainer[];
+    addToBundleIdentifier: string;
+    actionHeaderOptions: ActionHeaderOptions;
+    tableColumns: DataTableColumn[];
+    isEnterprise: boolean;
+    hasEnvironments: boolean;
+    stateLabels: { [key: string]: string };
+}
+
+@Injectable()
+export class DotContainerListStore extends ComponentStore<DotContainerListState> {
+    constructor(
+        private route: ActivatedRoute,
+        private dotMessageService: DotMessageService,
+        private dotRouterService: DotRouterService,
+        private dotPushPublishDialogService: DotPushPublishDialogService
+    ) {
+        super(null);
+
+        this.init();
+
+        this.route.data
+            .pipe(pluck('dotContainerListResolverData'), take(1))
+            .subscribe(([isEnterprise, hasEnvironments]: [boolean, boolean]) => {
+                this.updateResolvedProperties([isEnterprise, hasEnvironments]);
+                this.setContainerBulkActions();
+            });
```

Comment:
Here you are updating the state three times and that means 3 change detection, you can do this in one pass.